controller/specialized/aigateway_controller_reconciler_owned_resources.go

```diff
@@ -194,6 +197,15 @@ func (r *AIGatewayReconciler) configurePlugins(
 	if aiGateway.Spec.CloudProviderCredentials.Namespace != nil {
 		credentialSecretNamespace = *aiGateway.Spec.CloudProviderCredentials.Namespace
 	}
+
+	// check if referencing the credential secret is allowed by referencegrants.
+	if !r.secretReferenceAllowedByReferenceGrants(ctx, logger, aiGateway, credentialSecretNamespace, credentialSecretName) {
+		log.Info(logger, "Referencing Secret is not allowed by ReferenceGrants",
+			"secret_namespace", credentialSecretNamespace, "secret_name", credentialSecretName)
```

Comment:
Given we're returning the error from here, I don't think a separate `Info` log is needed. The controller runtime will log the error, right?
```suggestion
```

Reply:
Done.